server/aesdsocket.c

```diff
@@ -95,6 +97,10 @@ static int process_msg(char *msg, int fd, pthread_mutex_t *mutex)
 #if (USE_AESD_CHAR_DEVICE == 0)
     struct stat statbuf;
     off_t offset = 0;
+#elif (USE_AESD_CHAR_DEVICE == 1)
+    regex_t preg;
+    char *pattern = "(AESDCHAR_IOCSEEKTO).*";
```

Comment:
@hgajapathy unique idea to use regex here.  You could probably match on something more specific to the pattern we are using with commas and numeric digits.